include/picongpu/fields/incidentField/profiles/GaussianPulse.hpp

```diff
@@ -225,10 +225,14 @@ namespace picongpu::fields::incidentField
                     // the front of the laser pulse.
                     constexpr auto timeshift = LongitudinalEnvelope::TIME_SHIFT;
                     time += timeshift;
-                    // calculate focus position relative to the current point in the propagation direction
+                    /* Calculate focus position relative to the current point in the propagation direction.
+                     * Coordinate system is PIConGPUs and not the laser internal coordinate system where X is the
+                     * propagation direction.
+                     */
                     auto const focusRelativeToOrigin = this->focus - this->origin;
 
-                    float_X const focusPos = math::sqrt(pmacc::math::l2norm2(focusRelativeToOrigin)) - pos[0];
+                    // Distance from the current cell to the focus in laser propagation direction.
+                    float_X const focusPos = focusRelativeToOrigin.sumOfComponents() - pos[0];
```

Comment:
I am not sure whether this still works if less than 2 of the 3 components are zero.